app/main.py

```diff
@@ -1,214 +1,56 @@
-KNIGHTS = {
-    "lancelot": {
-        "name": "Lancelot",
-        "power": 35,
-        "hp": 100,
-        "armour": [],
-        "weapon": {
-            "name": "Metal Sword",
-            "power": 50,
-        },
-        "potion": None,
-    },
-    "arthur": {
-        "name": "Arthur",
-        "power": 45,
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
+from app.Knights.knight import Knight
+from app.Staff.armour import Armour
+from app.Staff.potion import Potion
+from app.Staff.weapon import Weapon
+
+
+def knights_creation(config: dict) -> Knight:
+    character = Knight(config["name"], config["power"], config["hp"])
+    armours = []
+    if config["armour"] is not None:
+        for armour in config["armour"]:
+            armours.append(Armour(
+                armour.get("part"),
+                armour.get("protection")
+            ))
+    if config["potion"] is not None:
+        effect = config["potion"]["effect"]
+        potion = Potion(
+            config["potion"]["name"],
+            effect.get("power", 0),
+            effect.get("hp", 0),
+            effect.get("protection", 0)
+        )
+    else:
+        potion = Potion("Unknown")
+    weapon = Weapon(config["weapon"]["name"], config["weapon"]["power"])
+    character.knight_stats_completing(armours, weapon, potion)
+
+    return character
```

Comment:
The function `knights_creation` is responsible for creating a Knight object and equipping it with armour, a potion, and a weapon. Consider extracting the potion creation logic into a separate function to adhere to the DRY principle, as similar logic might be reused elsewhere.